Update argument scores

nickkeesG opened the pull request:
Split getArgumentScores into one for each score. Each score calculation ONLY considers users who voted AND have a relevant ingroup/outgroup. This should fix the issue we had with unusually low scores. 

@sofvanh

nickkeesG committed:
refactored argumentScoreHandler

nickkeesG committed:
fixed conflict with main

sofvanh committed:
Merge branch 'main' into updateArgumentScores

sofvanh commented:
@nickkeesG Something seems wrong with the calculation of Fragmentation. A lot of the nodes that should score high in fragmentation are showing "Not enough data to calculate scores".

CEALLAR Dec 4 on this branch:
<img width="591" alt="Screenshot 2024-12-18 at 15 31 58" src="https://github.com/user-attachments/assets/7708208e-f81a-4df6-993a-c1fd669419bb" />

CEALLAR Dec 4 on main:
<img width="597" alt="Screenshot 2024-12-18 at 15 31 53" src="https://github.com/user-attachments/assets/05ab7d34-cbe5-4615-8c44-43436104f1f0" />

nickkeesG commented:
There are two changes which need to happen in the ArgumentScoreHandler:

1. I made a mistake in getFragmentationScore. The final return statement should be multiplied by two
return 2 * fragmentationSum / uniquenessSum     // normalize score to range 0 - 1

2. Currently, a node can only get a fragmentation score if it ALSO gets a consensus score. I recommend we allow for nodes to receive scores, even if the other one isn't computable yet. There's an if statement at the end of argumentScoreHandler which currently requires neither score be null. 

This is what is currently causing the graph to look way less orange than it should.

sofvanh commented:
I'll think about how to deal with this - probably we'll show frag/cons even if the other one isn't being calculated.

What happened here is that the nodes that don't show a score on the branch were mostly voted on by people who are friends with one another, so the nodes have a fragmentation score but not a consensus score.

sofvanh committed:
Merge branch 'main' into updateArgumentScores

sofvanh commented:
Solution: Allow consensus and fragmentation to be undefined, in which case show "More data needed" in score info.

![Screenshot 2025-01-07 at 17 52 51](https://github.com/user-attachments/assets/b454017b-fc30-4960-bfeb-1ac31bcb8e0d)

sofvanh commented:
Notice how with the new scoring algorithm we get more bright yellows!

sofvanh committed:
Fix bugs&formatting in argumentScoreHandler

Add "2 *" to getFragmentationScore
Change `Score` type to allow undefined Consensus and Fragmentation
